pkg/storage/unified/apistore/prepare.go

```diff
@@ -71,9 +73,20 @@ func (s *Storage) prepareObjectForUpdate(ctx context.Context, updateObject runti
 	if err != nil {
 		return nil, err
 	}
+
+	if previous.GetUID() == "" {
+		return nil, fmt.Errorf("object is missing UID")
+	}
+
 	obj.SetUID(previous.GetUID())
 	obj.SetCreatedBy(previous.GetCreatedBy())
 	obj.SetCreationTimestamp(previous.GetCreationTimestamp())
+	obj.SetResourceVersion("")
+	obj.SetName(previous.GetName())
```

Comment:
^^^ this feels more like an error case!

If the names do not match, something very very bad happened

Reply:
agreed

Reply:
makes sense, changing it!